src/main/plugins/spellcheck-plugin/spellcheck-plugin.ts

```diff
@@ -0,0 +1,104 @@
+import { ExecutionPlugin } from "../../execution-plugin";
+import { PluginType } from "../../plugin-type";
+import { SpellcheckOptions } from "../../../common/config/spellcheck-options";
+import { SearchResultItem } from "../../../common/search-result-item";
+import { UserConfigOptions } from "../../../common/config/user-config-options";
+import { TranslationSet } from "../../../common/translation/translation-set";
+import { SpellcheckSearcher } from "./spellcheck-searcher";
+import { defaultSpellcheckIcon } from "../../../common/icon/default-icons";
+
+interface SpellcheckResult {
+    suggestion: string;
+}
+
+export class SpellcheckPlugin implements ExecutionPlugin {
+    public pluginType = PluginType.Dictionary;
```

Comment:
Each plugin should have it's own PluginType. For this I suggest `PluginType.Spellcheck`